File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/backend/connected/aas/submodelelement/property/ConnectedReferenceElement.java
package org.eclipse.basyx.aas.backend.connected.aas.submodelelement.property;

import org.eclipse.basyx.aas.api.metamodel.aas.reference.IReference;
import org.eclipse.basyx.aas.api.metamodel.aas.submodelelement.property.IReferenceElement;
import org.eclipse.basyx.aas.backend.connected.aas.submodelelement.ConnectedDataElement;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.submodelelement.ReferenceElement;
import org.eclipse.basyx.vab.core.proxy.VABElementProxy;
/**
 * "Connected" implementation of IReferenceElement
 * @author rajashek
 *
 */
public class ConnectedReferenceElement extends ConnectedDataElement implements IReferenceElement {
	public ConnectedReferenceElement(String path, VABElementProxy proxy) {
		super(path, proxy);		
	}
	

	@Override
	public void setValue(IReference ref) {
		getProxy().updateElementValue(constructPath(ReferenceElement.VALUE), ref);
		
	}

	@Override
	public IReference getValue() {
		return (IReference)getProxy().readElementValue(constructPath(ReferenceElement.VALUE));
	}

}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/connected/submodelelement/property/ConnectedReferenceElement.cpp
/*
 * ConnectedReferenceElement.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/connected/submodelelement/property/ConnectedReferenceElement.h>

namespace basyx {
namespace submodel {

ConnectedReferenceElement::ConnectedReferenceElement(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedDataElement(proxy)
{}

void ConnectedReferenceElement::setValue(const IReference & ref)
{
  //todo
  //this->setProxyValue(submodelelement::property::PropertyPaths::VALUE, ref);
}

std::shared_ptr<IReference> ConnectedReferenceElement::getValue() const
{
  //todo 
  return nullptr;// this->getProxyValue(submodelelement::property::PropertyPaths::VALUE);
}

}
}

